academy/xl-run-prod/1-software.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Prepare the software to run
+order: 1
+description: Build and deploy
+tag: deep-dive
+---
+
+# Prepare the software to run
+
+To get to production, the first order of business is to build the binary that the nodes will run. If you used Ignite CLI, then you already did this under the hood with the `ignite chain serve` command.
+
+## Target platforms
+
+Because you are going to run the nodes on different machines, it is possible they will use different operating systems and CPU types. You need to account for that when building the binary. In particular, the computer on which you build may be entirely different from the one on which you eventually run the binary. In the jargon, you need to specify the target platform(s).
+
+What target platforms are available? Because you built your entire blockchain with Go, you can rely on [Go target platforms](https://go.dev/doc/install/source#environment) for that. Or a nicely presented one [such as this](https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63). To get the targets specific to your version of Go, just run:
+
+```sh
+$ go tool dist list
+```
+
+For instance, for `go version go1.18.3 linux/amd64`, it returns:
+
+```txt
+aix/ppc64
+android/386
+android/amd64
+android/arm
+android/arm64
+darwin/amd64
+darwin/arm64
+dragonfly/amd64
+freebsd/386
+freebsd/amd64
+freebsd/arm
+freebsd/arm64
+illumos/amd64
+ios/amd64
+ios/arm64
+js/wasm
+linux/386
+linux/amd64
+linux/arm
+linux/arm64
+linux/mips
+linux/mips64
+linux/mips64le
+linux/mipsle
+linux/ppc64
+linux/ppc64le
+linux/riscv64
+linux/s390x
+netbsd/386
+netbsd/amd64
+netbsd/arm
+netbsd/arm64
+openbsd/386
+openbsd/amd64
+openbsd/arm
+openbsd/arm64
+openbsd/mips64
+plan9/386
+plan9/amd64
+plan9/arm
+solaris/amd64
+windows/386
+windows/amd64
+windows/arm
+windows/arm64
+```
+
+As a side-note, some of these platforms are first class ports of Go while the others are not. If you want to only see the first class ports and have installed the `jq` tool, you can run:
+
+```sh
+$ go tool dist list -json | jq -r '.[] | select(.FirstClass) | [.GOOS , .GOARCH] | join("/")'
+```
+
+Notice the `GOOS` and `GOARCH` keywords in the command above. You will see them again later. The list is now much shorter:
+
+```txt
+darwin/amd64
+darwin/arm64
+linux/386
+linux/amd64
+linux/arm
+linux/arm64
+windows/386
+windows/amd64
+```
+
+Imagine you are going to run the node:
+
+* On regular office Linux boxes, so target `linux/amd64`.
+* And on AWS EC2 instances with a Graviton processor, so target `linux/arm64`.
+* Validator operators are going to generate their genesis transactions on Mac computers with Intel CPUs, so target `darwin/amd64`.
+
+## Build and package
+
```

Comment:
@xavierlepretre if possible add a sentence here to not have to headlines following each other.